### ISSUE TITLE
OAuth.clearCache() does not work as intended

### ISSUE THREAD

**Issue by angelxmoreno** [oauth-io/oauth-js#76]
looking at the docs: 
```js
//clears the cache for Facebook
OAuth.clearCache('facebook');
//removes cache for all providers
OAuth.clearCache();
```

however, looking at the source:
```js
clearCache: function(provider) {
    cookies.eraseCookie("oauthio_provider_" + provider);
}
```
and
```js
eraseCookie: function(name) {
    var date;
    date = new Date();
    date.setTime(date.getTime() - 86400000);
    this.document.cookie = name + "=; expires=" + date.toGMTString() + "; path=/";
  }
```
basically tells me that it is trying to set the cookie with a name of `oauthio_provider_undefined` and thus, not clearing the cookie at all!

**Comment by thyb** [comment]
Thanks a lot for digging into this issue. It's indeed a bug, we've just commit a fix for this issue (https://github.com/oauth-io/oauth-js/commit/069051aa2f226d46637234740de4560e8386d6e4), it will be merged in the v0.4.5 this week. Thanks again.

**Comment by angelxmoreno** [comment]
:+1:
